### PR TITLE
fix region error and grpc was cancelled

### PR DESCRIPTION
**Pull request by zhexuany**
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/146)

### CODE REVIEW

File: src/main/java/com/pingcap/tikv/policy/RetryPolicy.java [hunk 1]
```diff
@@ -46,13 +47,15 @@ private boolean checkNotRecoverableException(Status status) {
...
-  private void handleFailure(Exception e, String methodName, long millis) {
+  private void handleFailure(Exception e, String methodName, long nextBackMills) {
+    if(nextBackMills == BackOff.STOP) {
+          throw new GrpcException("retry is exhausted.", e);
+    }
     Status status = Status.fromThrowable(e);
     if (checkNotRecoverableException(status)) {
-      logger.error(String.format("Failed to recover from last grpc error calling %s.", methodName));
-      throw new GrpcException(e);
+      throw new GrpcUnrecoverableException(e);
```
Comment (ilovesoup) [inline src/main/java/com/pingcap/tikv/policy/RetryPolicy.java]: why do we need this extra exception type? We already have a method testing if unrecoverable.
I would do a simple re-throw.

File: src/main/java/com/pingcap/tikv/operation/KVErrorHandler.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public void handle(RespT resp) {
         regionManager.onRegionStale(ctxRegion.getId(), error.getStaleEpoch().getNewRegionsList());
         this.regionManager.onRegionStale(
             ctxRegion.getId(), error.getStaleEpoch().getNewRegionsList());
-        throw new StatusRuntimeException(Status.fromCode(Status.Code.CANCELLED).withDescription(error.toString()));
+        throw new StatusRuntimeException(Status.fromCode(Status.Code.UNAVAILABLE).withDescription(error.toString()));
```
Comment (ilovesoup) [inline src/main/java/com/pingcap/tikv/operation/KVErrorHandler.java]: why modified both. you removed cancelled status from unrecoverable and change it to unavailable.
I think whether an exception is recoverable or not completely depend on TiKV's definition. It's better to confirm with the team.

File: src/main/java/com/pingcap/tikv/policy/RetryPolicy.java (hunk 1 above)
Comment (zhexuany) [reply src/main/java/com/pingcap/tikv/policy/RetryPolicy.java]: trying to do two things in one PR. probably gonna do it in another PR.

**Review by ilovesoup — APPROVED** [review]
LGTM

**Pull request merged**